python/pyspark/sql/dataframe.py

```diff
@@ -271,11 +279,38 @@ def explain(self, extended=False):
         ...
         == Physical Plan ==
         ...
+
+        >>> df.explain(mode="formatted")
+        == Physical Plan ==
+        * Scan ExistingRDD (1)
+        <BLANKLINE>
+        <BLANKLINE>
+        (1) Scan ExistingRDD [codegen id : 1]
+        Output: [age#0, name#1]
+        <BLANKLINE>
+        <BLANKLINE>
+
+        .. versionchanged:: 3.0.0
+           Added optional argument `mode` to specify the expected output format of plans.
         """
-        if extended:
-            print(self._jdf.queryExecution().toString())
+        if extended is not None and mode is not None:
+            raise Exception("extended and mode can not be specified simultaneously")
+
+        # Handle a case with no argument first: df.explain()
```

Comment:
Yes .. this case is a bit annoying. There's an example to follow at `sample`.

Reply:
ok, I'll update based on the suggestion. Thanks.